### ISSUE TITLE
Update Vite v5.1

### ISSUE THREAD

**Issue by dai-shi** [dai-shi/waku#474]
I tried to upgrade Vite 5.1.1 in https://github.com/dai-shi/waku/pull/471, but it broke something related to styles.

**Comment by dai-shi** [comment]
Thanks. I also added a test: https://github.com/dai-shi/waku/pull/471/commits/81098083f9c033ff82d0d27f1a37f239ddc0f697
It only reproduces with standalone setup. I didn't see any issues in monorepo setup.

**Comment by campbellman** [comment]
tldr: fixing the websocket connection for hmr causes styles to work when hot reloaded, but _not_ when the browser is hard refreshed. Maybe a helpful hint?

More details and the commits to reproduce. I used the 02_demo example to test.

1. https://github.com/dai-shi/waku/commit/de220e3222c660c0f18f5f27dd7e2c00c0bc4209
  - **description**: upgrades vite to 5.1.1 everywhere
  - **observations**: css fails as expected when running `pnpm run dev` in 02_demo
2. https://github.com/dai-shi/waku/commit/b6627c86d1f6e9248d412b44677cf5103ad33f33
  - **description**: changes usages of `ViteDevServer.ws`=> `ViteDevServer.hot`, specifies a port in the dev handler's config at `server.hmr.port` that equals the DEV mode port + 1.
  - **why**: was curious if changing waku's usage of the depreciated `ViteDevServer.ws` interface would help. And the `server.hmr.port` change addresses the dev mode error: "WebSocket server error: Port is already in use" that I was seeing
  - **observations**: (see attached video) something interesting happens after this change - to reproduce, go to 02_demo, run `pnpm run dev`, and hard refresh http://localhost:3000 to observe that the styling still doesn't work.  Then, without closing your browser tab, open `02_demo/src/templates/root-layout.tsx` file in your IDE and hit save to trigger a hot reload. If you navigate back to the browser running the example you'll observe that the styles show up. Next, hard refresh the tab and observe that the styles are once again gone.

https://github.com/dai-shi/waku/assets/97706003/21ef47c7-ccb3-4355-b394-8806d081950d

**Comment by Aslemammad** [comment]
going to work on this

**Comment by dai-shi** [comment]
> Maybe a helpful hint?

I'm pretty sure @Aslemammad gets a hint from it!

> "WebSocket server error: Port is already in use" that I was seeing

I was curious about it too. Is `ws` already deprecated in Vite 5?

**Comment by campbellman** [comment]
@dai-shi yes it appears so. I wasn't able to find documentation for it but here's the vite PR that marks `ws` as depreciated https://github.com/vitejs/vite/pull/15340/files#diff-abb3345b6e3b2ec6d297c2ebc54cca85ae4487a31bac3cc9e78457f5114adb26R235.

Also, I have a third commit in addition to the two above that _appears_ to fix this issue: https://github.com/dai-shi/waku/commit/7d7d8bf5e0719d60745bfbe3ddbc6bdc6f7efba2.

I'm not confident about these changes but wanted to share anyways so that someone can run it and take a look too. I'll try to provide more details in a day or two if this issue is still open @Aslemammad
